drivers/adc/ad7606/ad7606.h

```diff
@@ -391,5 +391,7 @@ int32_t ad7606_set_digital_diag(struct ad7606_dev *dev,
 				struct ad7606_digital_diag diag);
 int32_t ad7606_init(struct ad7606_dev **device,
 		    struct ad7606_init_param *init_param);
+int32_t ad7606_data_correction_serial(struct ad7606_dev *dev, 
```

Comment:
i assume this new function is called after getting the samples with ad7606_read_samples(), i'd suggest to document this sequence somewhere

another alternative would be to make this new function private and propose a function that gets the data and performs the splitting into data and status buffers internally, perhaps called ad7606_read_data_status()

Reply:
hi,
  I prefer former solution because even creating new function, a new pointer for status buffer is still requires, this will change the parameter list of original data read function `ad7606_read_samples()`, which affects all exist project. I feel this is redundent for call 2 functions in a sequence but cannot figure out a better way. I believe this happens a lot during a Build-and-Fix develop mode.
 so where can I do the documentation? I find there is no readme or anything else for single device SDK, perhaps just point out in comments that could be called after read functions?

Reply:
So what I meant by the 2nd proposal is that you'd have both these functions:
```
ad7606_read_data() // existing, no changes
ad7606_read_data_status() // new

// private/static, inaccessible to the device driver user
ad7606_data_correction_serial()

// the new function makes data reading into data/status buffers "atomic" in the sense that it's only one operation and the user doesn't need to check for intermediate status codes or handle intermediate buffers
ad7606_read_data_status()
     call ad7606_read_data()
     calls ad7606_data_correction_serial() with the output of ad7606_read_data()
```

I hope this is more clear, let me know which approach you choose.

Reply:
I chose the 1nd suggestion and have added a note in the funciton description part, thx
![image](https://github.com/user-attachments/assets/5288ddea-04e2-47e0-b3fb-44c34595db5c)